### ISSUE TITLE
[BUG] Rancher integration with Harvester 

### ISSUE THREAD

**Issue by PolarisPT** [harvester/harvester#7036]
**Describe the bug**
I have a harvester hci  1.3.2 (single node - 10.101.58.19 - VIP 10.101.58.20)  and a rancher instance (a vm inside harvester with k3s  rancher 2.9.3 - ip 10.101.58.27) both on the same network  10.101.58.0/24 (cluster network).
Harvester has the following proxy config :
{
  "httpProxy": "http://xxxyyypx01.aaa.com:3128",
  "httpsProxy": "http://xxxyyypx01.aaa.com:3128",
  "noProxy": "localhost,127.0.0.1,::1,10.101.58.19,10.101.58.20,10.101.58.27,10.101.58.0/24,.aaa.com,xxxyyypx01.aaa.com"
}
Rancher was deployed with this helm config with an internally selfsign certificate from aaa.com for xxxinovdevrch1.aaa.com:
"
 helm install rancher rancher-latest/rancher \
    --namespace cattle-system \
    --set proxy=http://xxxyyypx01.aaa.com:3128 \
    --set noProxy=127.0.0.0/8\\,10.0.0.0/8\\,cattle-system.svc\\,172.16.0.0/12\\,192.168.0.0/16\\,.svc\\,.cluster.local\\,10.101.58.0/24\\,.aaa.com \
    --set hostname=xxxinovdevrch1.aaa.com\
    --set bootstrapPassword=admin \
    --set ingress.tls.source=secret \
    --set replicas=1 \
    --version 2.9.2 \
    --set useBundledSystemChart=true
"
When i generate a registration URL in rancher and added it to harvester  :
"
https://xxxinovdevrch1.aaa.com/v3/import/2f7ltdgjtjb65skchkx698ltxrgpdv86jwqfwnlw7nswfc49cpmrqj_c-m-ztnrdlv6.yaml
"
I get an error :
"
Get "https://xxxinovdevrch1.aaa.com/v3/import/2f7ltdgjtjb65skchkx698ltxrgpdv86jwqfwnlw7nswfc49cpmrqj_c-m-ztnrdlv6.yaml": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
"

When i log into my harvester node ( 10.101.58.19) via ssh with rancher user and execute a curl to check network connectivity the curl returns ok:

So i'm  assuming this is an error/bug from harvester internal network (canal??)
How can i debug this issue further ?

**To Reproduce**
1. Go to rancher, virtualization management, import existing, create
2. grab registration URL and paste it into harvester, settings, cluster-registration-url

**Expected behavior**
I would expect harvester to deploy the manifests inside the registration yaml, namely the deployment [cattle-cluster-agent].
And after that integration in racher to change from "pending" to "active" so that i can provision rke2 cluster using harvester cloud provider.

**Support bundle**

**Environment**
 - Harvester ISO version: 1.3.2
 - Underlying Infrastructure Baremetal with PowerEdge R650xs

**Additional context**

**Comment by starbops** [comment]
Hi @PolarisPT, do you mean there is no issue curling `https://xxxinovdevrch1.aaa.com/v3/import/2f7ltdgjtjb65skchkx698ltxrgpdv86jwqfwnlw7nswfc49cpmrqj_c-m-ztnrdlv6.yaml` directly from the Harvester node?

It would be nice to have a support bundle attached so that we can look into the situation. Thank you.

**Comment by PolarisPT** [comment]
Hi @starbops, support bundle sent via email.

Thanks!

**Comment by PolarisPT** [comment]
> Hi @PolarisPT, do you mean there is no issue curling `https://xxxinovdevrch1.aaa.com/v3/import/2f7ltdgjtjb65skchkx698ltxrgpdv86jwqfwnlw7nswfc49cpmrqj_c-m-ztnrdlv6.yaml` directly from the Harvester node?
> 
> It would be nice to have a support bundle attached so that we can look into the situation. Thank you.

That's correct, no issues executing the curl from the harvester node, the issue seems to happen only from pods/containers on the harvester rke2.

Also sent support bundle to harvester-support-bundle@suse.com

Thanks

**Comment by starbops** [comment]
Could you check the `net.bridge.bridge-nf-call-iptables` kernel tunable?

```
sysctl net.bridge.bridge-nf-call-iptables
```

If it is `1`, please try to turn it off:

```
sysctl -w net.bridge.bridge-nf-call-iptables=0
```

And see if the issue remains or not.
Thank you.

**Comment by PolarisPT** [comment]
Hi,

Thanks for your quick response...
After executing 

```
sysctl -w net.bridge.bridge-nf-call-iptables=0
```

Now harvester can download the rancher cluster agent manisfests just fine, thanks..

So the issue is with the iptables, any clue why ?

Attaching the full output from "iptables -L" from harvester node.

Thanks!
[iptables.txt](https://github.com/user-attachments/files/17947206/iptables.txt)

**Comment by Hector-Work** [comment]
I have exactly the same problem, except that for me `net.bridge.bridge-nf-call-iptables` is already `0`.

**Comment by starbops** [comment]
@PolarisPT Your case is similar to #4359. Basically, VLAN-tagged packets flow between the harvester pod and the rancher VM is altered prematurely when they arrived the Linux bridge. They should be handled properly at the VLAN interface, otherwise, the receiver will see VLAN tagged packets which it cannot handle.

**Comment by starbops** [comment]
Hi @Hector-Work do you have a support bundle? It might be different cases, but better to look at the support bundle first.

**Comment by Hector-Work** [comment]
Thanks @starbops, I have since redeployed and it's working as expected now, so unfortunately I can't help with debugging this.

**Comment by PolarisPT** [comment]
> @PolarisPT Your case is similar to #4359. Basically, VLAN-tagged packets flow between the harvester pod and the rancher VM is altered prematurely when they arrived the Linux bridge. They should be handled properly at the VLAN interface, otherwise, the receiver will see VLAN tagged packets which it cannot handle.

Hi, i'm happy with the workaround for now...

```
sysctl -w net.bridge.bridge-nf-call-iptables=0
````

But i'm worried about the lack of persistence of this change, is there a way to make this fix survive a server restart ?

Thanks

**Comment by w13915984028** [comment]
@PolarisPT You may follow https://github.com/harvester/harvester-installer/pull/884 to manually persistant the change.

**Comment by PolarisPT** [comment]
thx, closing
